File: hnaj-be/app/Http/Controllers/Api/Auth/MeController.php
<?php

namespace App\Http\Controllers\Api\Auth;

use App\Actions\Auth\LoadAuthenticatedUser;
use App\Http\Controllers\Controller;
use App\Http\Resources\UserResource;
use App\Http\Responses\ApiResponse;
use Illuminate\Http\JsonResponse;
use Illuminate\Http\Request;

class MeController extends Controller
{
    public function __invoke(Request $request, LoadAuthenticatedUser $loadAuthenticatedUser): JsonResponse
    {
        return ApiResponse::success(
            data: ['user' => new UserResource($loadAuthenticatedUser->handle($request->user()))],
            message: 'Authenticated user loaded successfully.',
        );
    }
}

File: hnaj-be-java/src/main/java/com/hnaj/auth/controller/MeController.java
package com.hnaj.auth.controller;

import com.hnaj.api.common.ApiResponse;
import com.hnaj.auth.AuthenticatedUser;
import com.hnaj.auth.AuthRoleService;
import com.hnaj.auth.UserView;
import com.hnaj.auth.security.RequireRole;
import java.util.Map;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class MeController {
    private final AuthRoleService roleService;

    public MeController(AuthRoleService roleService) {
        this.roleService = roleService;
    }

    @GetMapping("/api/auth/me")
    @RequireRole({"user", "sub_admin"})
    public ResponseEntity<Map<String, Object>> me(AuthenticatedUser principal) {
        UserView view = UserView.from(principal.user(),
                roleService.namesOf(principal.user().getId()));
        return ApiResponse.success(Map.of("user", view),
                "Authenticated user loaded successfully.", null);
    }
}